datalad/support/annexrepo.py

```diff
@@ -2144,7 +2148,14 @@ def _run_annex_command_json(self, command,
             annex_options += ['--json-progress']
 
         if jobs == 'auto':
-            jobs = N_AUTO_JOBS
+            # Limit to # of CPUs (but at least 3 to start with)
+            # and also an additional config constraint (by default 1
+            # due to https://github.com/datalad/datalad/issues/4404)
+            jobs = self._n_auto_jobs or min(
+                self.config.obtain('datalad.runtime.max-annex-jobs'),
+                max(3, cpu_count()))
```

Comment:
why bother limiting here more from what user requested? I could have 1 cpu but download in parallel could still be beneficial (it is not CPU bound job).  I would just pass the value of the configuration here.
Although indeed it kinda deminishes the notion of any "auto" here -- it pretty much becomes "config", it is worth adjusting documentation where jobs is mentioned that it wouldn't be "auto" but rather consult config variable, or better change from "auto" to "config", and then here leave it all as it was (i.e. for "auto" - do previous behavior).  So by default it would consult config - would become 1.

Reply:
This way, later, if we could come up with some true "auto" we could change back while leaving possibility to still use "config" setting.